src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -37,20 +37,47 @@ class MessageBufferProcessor(
 	messageLockTimeOutAfter: FiniteDuration
 ) {
 
-	@SuppressWarnings(Array("org.brianmckenna.wartremover.warts.Var"))
-	private var resendSchedule: Option[Cancellable] = None
+	private case class StartSchedule(ec: ExecutionContext)
+	private case object StopSchedule
+	private case class StopLocking(ec: ExecutionContext)
+	private case class ResumeLocking(ec: ExecutionContext)
+
+	private val resendSchedule = actorSystem.actorOf(Props(new Actor {
+
+		def createSchedule(lock: Boolean)(implicit ec: ExecutionContext): Cancellable =
+			actorSystem.scheduler.schedule(initialDelay, bufferProcessInterval)(
+				processMessageBuffer(lock)
+			)
+
+		def receive = {
+			case StartSchedule(ec) =>
+				context.become(repeating(createSchedule(true)(ec)))
+		}
+
+		def repeating(resendSchedule: Cancellable): Receive = {
+			case StopSchedule =>
+				resendSchedule.cancel()
+				context.unbecome()
+			case StopLocking(ec) =>
+				resendSchedule.cancel()
+				context.become(repeating(createSchedule(false)(ec)))
```

Comment:
Instead of repeatedly doing noop, you could just put the actor's state machine into a noop/sleeping state.

Reply:
It's not noop.

Reply:
We are stopping consumers, but we haven't yet. So, it makes sense to continue handling confirmations.

Reply:
Ah, I missed the top part of that schedule.